server/fleet/integrations.go

```diff
@@ -378,13 +381,35 @@ func ValidateGoogleCalendarIntegrations(intgs []*GoogleCalendarIntegration, inva
 		invalid.Append("integrations.google_calendar", "integrating with >1 Google Workspace service account is not yet supported.")
 	}
 	for _, intg := range intgs {
-		intg.Email = strings.TrimSpace(intg.Email)
-		if intg.Email == "" {
-			invalid.Append("integrations.google_calendar.email", "email is required")
+		if email, ok := intg.ApiKey[GoogleCalendarEmail]; !ok {
+			invalid.Append(
+				fmt.Sprintf("integrations.google_calendar.api_key_json.%s", GoogleCalendarEmail),
+				fmt.Sprintf("%s is required", GoogleCalendarEmail),
+			)
+		} else {
+			email = strings.TrimSpace(email)
+			intg.ApiKey[GoogleCalendarEmail] = email
```

Comment:
Nit: Should it modify `intg` if `email == ""`?.

Reply:
(Same with `privateKey` below.)

Reply:
If email is blank, we will fail validation, and intg won't be saved. So, it doesn't matter if we modify intg in that case.